ddtrace/contrib/internal/google_generativeai/_utils.py

```diff
@@ -0,0 +1,217 @@
+import sys
+
+import wrapt
+
+from ddtrace.internal.utils import get_argument_value
+
+
+class BaseTracedGenerateContentResponse(wrapt.ObjectProxy):
+    """Base wrapper class for GenerateContentResponse objects for tracing streamed responses."""
+
+    def __init__(self, wrapped, instance, integration, span, args, kwargs):
+        super().__init__(wrapped)
+        self._model_instance = instance
+        self._dd_integration = integration
+        self._dd_span = span
+        self._args = args
+        self._kwargs = kwargs
+
+
+class TracedGenerateContentResponse(BaseTracedGenerateContentResponse):
+    def __iter__(self):
+        try:
+            for chunk in self.__wrapped__.__iter__():
+                yield chunk
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            raise
+        else:
+            tag_response(self._dd_span, self.__wrapped__, self._dd_integration, self._model_instance)
```

Comment:
just for my own edification - i'm assuming `self.__wrapped__` is getting the accumulated chunks from the stream? is that all handled under the hood with `wrapt`? it just seems a bit different as to how we handle it in the `openai` integration.

Reply:
Yup, the google gemini SDK actually collects the chunks and joins them together into a regular response once the stream is finished, so we don't need to do any of the joining fortunately 😄